2025-03-16 16:25:54,803 : INFO : Start of Prediction
2025-03-16 16:25:54,803 : INFO : run_id:2025-03-16_162554_912588586
2025-03-16 16:25:54,804 : INFO : Start of Data Load, validation and transformation
2025-03-16 16:25:54,804 : INFO : Start of Archiving Old Rejected Files...
2025-03-16 16:25:54,804 : INFO : End of Archiving Old Rejected Files...
2025-03-16 16:25:54,805 : INFO : Start of Archiving Old Validated Files...
2025-03-16 16:25:54,809 : INFO : End of Archiving Old Validated Files...
2025-03-16 16:25:54,810 : INFO : Start of Archiving Old Processed Files...
2025-03-16 16:25:54,821 : INFO : End of Archiving Old Processed Files...
2025-03-16 16:25:54,822 : INFO : Start of Archiving Old Result Files...
2025-03-16 16:25:54,827 : INFO : End of Archiving Old Result Files...
2025-03-16 16:25:54,827 : INFO : Start of Reading values From Schema...
2025-03-16 16:25:54,827 : INFO : End of Reading values From Schema...
2025-03-16 16:25:54,828 : INFO : Start of Validating Column Length...
2025-03-16 16:25:54,828 : INFO : End of Validating Column Length...
2025-03-16 16:25:54,828 : INFO : Start of Validating Missing Values...
2025-03-16 16:25:54,829 : INFO : End of Validating Missing Values...
2025-03-16 16:25:54,829 : INFO : Start of Replacing Missing Values with NULL...
2025-03-16 16:25:54,829 : INFO : End of Replacing Missing Values with NULL...
2025-03-16 16:25:54,829 : INFO : Start of Creating Table...
2025-03-16 16:25:54,830 : INFO : Opened prediction database successfully
2025-03-16 16:25:54,856 : INFO : CREATE TABLE prediction_raw_data_t column_name
2025-03-16 16:25:54,863 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,872 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,879 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,888 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,897 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,906 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,914 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,924 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-16 16:25:54,925 : INFO : End of Creating Table...
2025-03-16 16:25:54,926 : INFO : Opened prediction database successfully
2025-03-16 16:25:54,927 : INFO : Start of Inserting Data into Table...
2025-03-16 16:25:54,927 : INFO : End of Inserting Data into Table...
2025-03-16 16:25:54,928 : INFO : Start of Exporting Data into CSV...
2025-03-16 16:25:54,929 : INFO : Opened prediction database successfully
2025-03-16 16:25:54,932 : INFO : End of Exporting Data into CSV...
2025-03-16 16:25:54,933 : INFO : Start of Moving Processed Files...
2025-03-16 16:25:54,933 : INFO : End of Moving Processed Files...
2025-03-16 16:25:54,934 : INFO : End of Data Load, validation and transformation
2025-03-16 16:25:54,934 : INFO : Start of Preprocessing...
2025-03-16 16:25:54,934 : INFO : Start of reading dataset...
2025-03-16 16:25:54,943 : INFO : End of reading dataset...
2025-03-16 16:25:54,943 : INFO : Start of feature encoding...
2025-03-16 16:25:54,982 : INFO : End of feature encoding...
2025-03-16 16:25:54,982 : INFO : Start of Droping Columns...
2025-03-16 16:25:54,984 : INFO : End of Droping Columns...
2025-03-16 16:25:54,984 : INFO : Start of finding missing values...
2025-03-16 16:25:54,985 : INFO : End of finding missing values...
2025-03-16 16:25:54,985 : INFO : Start of building final predictset...
2025-03-16 16:25:54,990 : INFO : End of building final predictset...
2025-03-16 16:25:54,990 : INFO : End of Preprocessing...
2025-03-16 16:25:54,990 : INFO : Start of Load Model
2025-03-16 16:25:54,991 : INFO : Model File KMeans loaded
2025-03-16 16:25:54,991 : INFO : End of Load Model
2025-03-16 16:25:54,994 : ERROR : Unsuccessful End of Prediction
Traceback (most recent call last):
  File "C:\Users\KIIT\PycharmProjects\PythonProject\apps\prediction\predict_model.py", line 60, in batch_predict_from_model
    clusters = kmeans.predict(self.X.drop(['empid'],axis=1))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIIT\PycharmProjects\PythonProject\.venv\Lib\site-packages\sklearn\cluster\_kmeans.py", line 1085, in predict
    X = self._check_test_data(X)
        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIIT\PycharmProjects\PythonProject\.venv\Lib\site-packages\sklearn\cluster\_kmeans.py", line 944, in _check_test_data
    X = validate_data(
        ^^^^^^^^^^^^^^
  File "C:\Users\KIIT\PycharmProjects\PythonProject\.venv\Lib\site-packages\sklearn\utils\validation.py", line 2944, in validate_data
    out = check_array(X, input_name="X", **check_params)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIIT\PycharmProjects\PythonProject\.venv\Lib\site-packages\sklearn\utils\validation.py", line 1130, in check_array
    raise ValueError(
ValueError: Found array with 0 sample(s) (shape=(0, 9)) while a minimum of 1 is required by KMeans.
